src/libsystemd/sd-json/sd-json.c

```diff
@@ -54,26 +57,26 @@ typedef struct JsonSource {
 } JsonSource;
 
 /* On x86-64 this whole structure should have a size of 6 * 64 bit = 48 bytes */
-struct JsonVariant {
+struct sd_json_variant {
         union {
                 /* We either maintain a reference counter for this variant itself, or we are embedded into an
                  * array/object, in which case only that surrounding object is ref-counted. (If 'embedded' is false,
                  * see below.) */
                 unsigned n_ref;
 
-                /* If this JsonVariant is part of an array/object, then this field points to the surrounding
+                /* If this sd_json_variant is part of an array/object, then this field points to the surrounding
                  * JSON_VARIANT_ARRAY/JSON_VARIANT_OBJECT object. (If 'embedded' is true, see below.) */
-                JsonVariant *parent;
+                sd_json_variant *parent;
         };
 
         /* If this was parsed from some file or buffer, this stores where from, as well as the source line/column */
         JsonSource *source;
         unsigned line, column;
 
-        /* The current 'depth' of the JsonVariant, i.e. how many levels of member variants this has */
+        /* The current 'depth' of the sd_json_variant, i.e. how many levels of member variants this has */
         uint16_t depth;
 
-        JsonVariantType type:8;
+        int type:8; /* actually sd_json_variant_type_t, but reduced to 8bit size */
```

Comment:
`Bit field type of type int should have explicitly unsigned integral, explicitly signed integral, or enumeration type.`

https://github.com/systemd/systemd/security/code-scanning/2614

Reply:
dunno, sd_json_variant_type_t is actually always going to be mapped to a signed `int` by the compiler. But of course we know that in this context it never can be negative, we are only going to store valid sd_json_variant_type_t in this field, i.e. not -EINVAL or so. Hence, the code as it is now is safe. I can understand the tools reasoning to cmplain abou this.

I'd probably say that it's a false positive, and leave the code as is.

Reply:
or maybe we can add the `signed` before it.

Reply:
Can you give that a shot? It should become apparent in the PR if it is enough to fix it

Reply:
→ https://github.com/systemd/systemd/pull/33305